src/posix.mak

```diff
@@ -63,222 +90,131 @@ WARNINGS := -Wall -Wextra \
-BACK_FLAGS := -DDMDV2=1 -I$(ROOT) -I$(TK) -I$(C) -I.
-ROOT_FLAGS := -DDMDV2=1 -I$(ROOT)
-
-
-DMD_OBJS = \
-	access.o attrib.o \
-	cast.o \
-	class.o \
-	constfold.o cond.o \
-	declaration.o dsymbol.o \
-	enum.o expression.o func.o nogc.o \
-	id.o \
-	identifier.o impcnvtab.o import.o inifile.o init.o inline.o \
-	lexer.o link.o mangle.o mars.o module.o mtype.o \
-	cppmangle.o opover.o optimize.o \
-	parse.o scope.o statement.o \
-	struct.o template.o \
-	version.o utf.o staticassert.o \
-	entity.o doc.o macro.o \
-	hdrgen.o delegatize.o interpret.o traits.o \
-	builtin.o ctfeexpr.o clone.o aliasthis.o \
-	arrayop.o json.o unittests.o \
-	imphint.o argtypes.o apply.o sapply.o sideeffect.o \
-	intrange.o canthrow.o target.o nspace.o errors.o \
-	escape.o tokens.o globals.o
-
-ROOT_OBJS = \
-	rmem.o port.o man.o stringtable.o response.o \
-	aav.o speller.o outbuffer.o object.o \
-	filename.o file.o async.o checkedint.o
-
-GLUE_OBJS = \
-	glue.o msc.o s2ir.o todt.o e2ir.o tocsym.o \
-	toobj.o toctype.o toelfdebug.o toir.o \
-	irstate.o typinf.o iasm.o
+CFLAGS += $(DEBUGFLAGS)
+
+
+# Specific compiler flags
+########################################################################
+# Add extra flags for each category of source files.
+DMD_FLAGS  := -I$(ROOT) -Wuninitialized
+GLUE_FLAGS := -I$(ROOT) -I$(TK) -I$(C)
+BACK_FLAGS := -I$(ROOT) -I$(TK) -I$(C) -I.
+ROOT_FLAGS := -I$(ROOT)
 
-ifeq (osx,$(OS))
-    GLUE_OBJS += libmach.o scanmach.o
-else
-    GLUE_OBJS += libelf.o scanelf.o
-endif
 
-#GLUE_OBJS=gluestub.o
-
-BACK_OBJS = go.o gdag.o gother.o gflow.o gloop.o var.o el.o \
-	glocal.o os.o nteh.o evalu8.o cgcs.o \
-	rtlsym.o cgelem.o cgen.o cgreg.o out.o \
-	blockopt.o cg.o type.o dt.o \
-	debug.o code.o ee.o symbol.o \
-	cgcod.o cod5.o outbuf.o \
-	bcomplex.o aa.o ti_achar.o \
-	ti_pvoid.o pdata.o cv8.o backconfig.o \
-	divcoeff.o dwarf.o \
-	ph2.o util2.o eh.o tk.o strtold.o \
+# Source object files
+########################################################################
+# In alphabetical order ... Sadly $(wildcard *.c) cannot be used since
+# the source directories contains unused files, or is a mix of files not
+# used on all OS/ARCH. A possible adjustment would be to clean-up and
+# move Linux/Mac specific files to src_linux, src_mac and win32-files to
+# src_win32, etc for example.
+DMD_OBJS := \
+	access.o aliasthis.o apply.o argtypes.o	arrayop.o attrib.o \
+	builtin.o canthrow.o cast.o class.o clone.o cond.o constfold.o \
+	cppmangle.o ctfeexpr.o declaration.o delegatize.o doc.o dsymbol.o \
+	entity.o enum.o errors.o escape.o expression.o func.o globals.o \
+	hdrgen.o id.o identifier.o impcnvtab.o imphint.o import.o \
+	inifile.o init.o inline.o interpret.o intrange.o json.o lexer.o \
+	link.o macro.o mangle.o mars.o module.o mtype.o nogc.o nspace.o \
+	opover.o optimize.o parse.o sapply.o scope.o sideeffect.o \
+	statement.o staticassert.o struct.o target.o template.o tokens.o \
+	traits.o unittests.o utf.o version.o
```

Comment:
Naa I did try that. Using ```$(wildcard src/*.c)```, which is my default mode for getting files to build (files not being built are renamed or removed) lists to many files. Some are aimed for win32 other for Mac, which is not strange. Some files though are not listed at all in the build files from what I can see.

A nice clean-up would be to have separate directories, the generic ```src```, and then one for each of Linux, Mac and Win32. Then ```$(wildcard ...)``` works, and there will be no confusion whether a file is deprecated, disabled or not. Git handle renames without a problem so it would not cause to much of a discomfort.